patterns/vsi/variables.tf

```diff
@@ -146,6 +146,14 @@ variable "vsi_per_subnet" {
   default     = 1
 }
 
+variable "user_data" {
```

Comment:
The complex validation that you added at the root level really belongs here at the solution level. If you are going to validate that the map keys here match entries in the `vpcs` list, then you want to do it here at the entry point, and maybe also at the module level of this solution.

Also the description needs to be more detailed explaining to the user what this map is supposed to be (with the key being a VPC name from the `vpcs` variable, and the data being the yaml).

There is something else to mention here: the user_data is usually in yaml format. We need to test that yaml inside an object still works correctly from all deployement types (terraform CLI, schematics, projects) and to make sure the yaml is valid by time it is applied to the VSI.